libs/coin-evm/src/__tests__/unit/adapters/etherscan.unit.test.ts

```diff
@@ -19,6 +19,63 @@ describe("EVM Family", () => {
   describe("adapters", () => {
     describe("etherscan", () => {
       describe("etherscanOperationToOperations", () => {
+        it("should convert an etherscan-like smart contract creation operation (from their API) to a Ledger Live Operation", () => {
+          const etherscanOp: EtherscanOperation = {
+            blockNumber: "14923692",
+            timeStamp: "1654646570",
+            hash: "0xaa45b4858ba44230a5fce5a29570a5dec2bf1f0ba95bacdec4fe8f2c4fa99338",
+            nonce: "7",
+            blockHash: "0x8df71a12a8c06b36c06c26bf6248857dd2a2b75b6edbb4e33e9477078897b282",
+            transactionIndex: "27",
+            from: "0x9aa99c23f67c81701c772b106b4f83f6e858dd2e",
+            to: "",
+            value: "0",
+            gas: "6000000",
+            gasPrice: "125521409858",
+            isError: "0",
+            txreceipt_status: "1",
+            input:
+              "0xa9059cbb000000000000000000000000313143c4088a47c469d06fe3fa5fd4196be6a4d600000000000000000000000000000000000000000003b8e97d229a2d54800000",
+            contractAddress: "0x4969d9fd2542e71e6b3ea87be54ea9a736bcc4e9",
+            cumulativeGasUsed: "1977481",
+            gasUsed: "57168",
+            confirmations: "122471",
+            methodId: "0xa9059cbb",
+            functionName: "transfer(address _to, uint256 _value)",
+          };
+
+          const accountId = encodeAccountId({
+            type: "js",
+            version: "2",
+            currencyId: "ethereum",
+            xpubOrAddress: "0x9aa99c23f67c81701c772b106b4f83f6e858dd2e",
+            derivationMode: "",
+          });
+
+          const expectedOperation: Operation = {
+            id: "js:2:ethereum:0x9aa99c23f67c81701c772b106b4f83f6e858dd2e:-0xaa45b4858ba44230a5fce5a29570a5dec2bf1f0ba95bacdec4fe8f2c4fa99338-FEES",
+            hash: "0xaa45b4858ba44230a5fce5a29570a5dec2bf1f0ba95bacdec4fe8f2c4fa99338",
+            accountId,
+            blockHash: "0x8df71a12a8c06b36c06c26bf6248857dd2a2b75b6edbb4e33e9477078897b282",
+            blockHeight: 14923692,
+            recipients: [""],
+            senders: ["0x9AA99C23F67c81701C772B106b4F83f6e858dd2E"],
+            value: new BigNumber("7175807958762144"),
+            fee: new BigNumber("7175807958762144"),
+            date: new Date("2022-06-08T00:02:50.000Z"),
+            transactionSequenceNumber: 7,
+            hasFailed: false,
+            nftOperations: [],
+            subOperations: [],
+            type: "FEES",
+            extra: {},
+          };
+
+          expect(etherscanOperationToOperations(accountId, etherscanOp)).toEqual([
```

Comment:
Not blocking but `toMatchInlineSnapshot` could be used here rather than manually creating and defining the whole `expectedOperation` object (yes I know we have this manually defined expected object pattern king of everywhere in the codebase, is it optimal? I don't know)
cf. https://jestjs.io/docs/snapshot-testing#inline-snapshots and https://jestjs.io/docs/expect#tomatchinlinesnapshotpropertymatchers-inlinesnapshot

Reply:
Personally I'm not fan of snapshots, you end up updating them everytime you make a change
It adds a lot of line but yeah it's a personal preference

Reply:
All solutions have pros & cons. 
I agree that snapshots are definitely not an amazing solution, it's faster but it's not in the test so you just never read it. 
Fixtures are little bit better as they're easier to access, and we have a lot of those already, but you still don't clearly see what you're testing right away as most of it is generated for you and generally contains a **lot** of fake data in it.

Clearly for a test for an adapter like this where you want to test that A — will become —> B, I think verbose is the best option. You want to see exactly what *should* be done directly, you're safe from any fake data or mock, and you're still as safe as if you were using fixtures since the typing is still added with TS.

Reply:
> I agree that snapshots are definitely not an amazing solution, it's faster but it's not in the test so you just never read it.

I agree with that and that is why I'm talking about **inline snapshots** (rather than the _normal_ snapshot we already use in some tests, which end up in a different file no one ever looks at).

I think inline snapshot provides the best of both worlds by writing the expected result inside the specific test case *and* being overall easier to write (generate) and maintain. Worth giving it a shot if you haven't played with it yet.

But at the end of the day, it's more or less the same and the result is the same (do we have a valid test that checks what we want). It's more a matter of personal preferences and ease of creating and maintenance.